core/src/plugins/filed/python/postgresql/bareos-fd-postgresql.py

```diff
@@ -764,12 +815,70 @@ def __do_switch_wal(self):
         Let PostgreSQL write latest transaction into a new WAL file now
         """
         try:
-            self.db_con.run("select pg_switch_wal()")
+            lsn = self.db_con.run("select pg_switch_wal()")[0][0]
+            bareosfd.JobMessage(
+                bareosfd.M_INFO,
+                f"pg_switch_wal to LSN '{lsn}'\n",
+            )
         except pg8000.Error as pg_err:
             bareosfd.JobMessage(
                 bareosfd.M_WARNING,
                 f"Could not switch to next WAL segment: {pg_err}\n",
             )
+        return lsn
+
+    def __decode_lsn_filename_offset(self, lsn):
+        """
+        Let PostgreSQL describe filename and bytes offset for the given lsn
+        if decoded we set self.wal_filename and self.wal_offset
```

Comment:
What does this function return ? What does it mean that the offset is 0 ?

Reply:
This function return True if the offset is 0.
if offset is 0, this mean that PostgreSQL actual lsn is a new file with no write in (which imply it doesn't have to be archived, and pg will skip archiving it, and as a consequence, we should not wait to see it happening in wal archive dir)

Reply:
We should probably add that comment to the doc string. Something like "returns True if .... and False otherwise" or "returns whether ...".

Reply:
I'm thinking of something like 
```
        """
        Let PostgreSQL describe filename and bytes offset for the given lsn.
        If decoded we set self.wal_filename and self.wal_offset and return True if the offset is
        zero (which mean we won't have to wait the corresponding wal, as it will not be archived)
        """

```